ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -20,12 +20,46 @@
 from abc import ABC, abstractmethod
 from typing import List, NamedTuple, Tuple, Optional, Union, Dict
 import numpy as np
-from enum import Enum
+from enum import Enum, IntEnum
 
 AgentId = int
 AgentGroup = str
 
 
+class EpisodeStatus(IntEnum):
+    """
+    The Episode is still ongoing
+    """
+
+    DEFAULT = 0
+
+    """
+    The Episode ended. The Agent either failed or succeeded at its task.
+    """
+    TERMINATED = 1
+
+    """
+    The Episode was interrupted. The Agent neither failed nor succeeded at the
+    task since it was interrupted.
+    """
+    INTERRUPTED = 2
+
+    def done(self) -> bool:
```

Comment:
It seems a little weird to me that we have these convenience methods - seems like they should match the actual enum states. I know they're for compat with the existing Python code - but let's say we just change that to use e.g. `if status == EpisodeStatus.TERMINATED`. Would we still want these?

Reply:
I would be happy to remove them to be honest. The issue would be that to replace the "done" behavior we have right now (either terminated OR interrupted), we would need to replace : 
```python
step_result.done[agent_id]
```
with
```python
step_result.status[agent_id] == EpisodeStatus.TERMINATED or step_result.status[agent_id] == EpisodeStatus.INTERRUPTED
```
or
```python
step_result.status[agent_id] in [EpisodeStatus.TERMINATED, EpisodeStatus.INTERRUPTED]
```
Other option could be to make it a IntFlag rather than IntEnum. Like suggested [here](https://docs.google.com/document/d/1qq3PzeNyWCvoGs_fxXKEneu3DFAMZGmPnPszTBeiw6M/edit#heading=h.f8mdsqs815me).

Reply:
You're right, that's pretty ugly. Do we imagine it's possible to have multiple of these things to be true? E.g. I both "won" and was "terminated" at the same time (tagging @andrewcoh) - I actually think it is possible.  

Also OK with just documenting the two helper methods. Would imagine a user of LL-API doesn't want to do that nasty equals every where either.

Reply:
We use the sign of the final reward to determine winning/losing. Not sure if that conflicts with this.

Reply:
It does not conflict since win/lose is not implemented with this. This is a **potential** "new" way of handling win/lose.
I think you can consider TERMINATED to always be true if INTERRUPTED is true if we use the flags option...

Reply:
We could have something like this for EpisodeStatus if we use IntFlag

```python
class EpisodeStatus(IntFlag):
    """
    The Episode is still ongoing
    """

    DEFAULT = 1

    """
    The Episode ended. The Agent either failed or succeeded at its task.
    """
    TERMINATED = 2

    """
    The Episode was interrupted. The Agent neither failed nor succeeded at the
    task since it was interrupted.
    """
    INTERRUPTED = 4
```

We could have our max_step be `EpisodeStatus.TERMINATED | EpisodeStatus.INTERRUPTED`.
But if we want to check if the Agent is "done" we would need something like:

```python
step_result.status[agent_id] & EpisodeStatus.TERMINATED == EpisodeStatus.TERMINATED

or 

EpisodeStatus.TERMINATED in step_result.status[agent_id]
```